sentry-android-core/src/test/java/io/sentry/android/core/util/MainThreadCheckerTest.kt

```diff
@@ -0,0 +1,28 @@
+package io.sentry.android.core.util
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlin.test.Test
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class MainThreadCheckerTest {
```

Comment:
`m`: Nice tests, would it make sense to add a test, that starts a thread, calls `MainThreadChecker.isMainThread()` inside it and makes sure it is false?

Reply:
it'd make sense, but as I tested them individually, not sure if it's any useful.

Reply:
I added a test for AppLifecycleIntegration which basically does that